### PR TITLE
Fixes slime heretics being hurt by rust

### LINKED ISSUE #56347: Rust Healing And Slime People Don't Seem To Mix
## Reproduction:

As explained by the person who said something about the bug:

1.  Be a slime person of any kind
2.  Also be a rust heretic, and have the ability that heals you on rust tiles
3.  Cry as you go to heal on your beloved rust tile, then die from the toxin healing, which hurts slime people

### PR DESCRIPTION
**Pull request by itseasytosee**
## About The Pull Request

fixes #56347

## Why It's Good For The Game

bugs
## Changelog
:cl: itseasytosee
fix:  Rust Heritec slime people will no longer be hurt by rust
/:cl:

### CODE REVIEW

**Comment by necromanceranne** [comment]
Could you do this fix for eldritch essence as well?

**Comment by itseasytosee** [comment]
Yeah sure

**Comment by Rohesie** [comment]
Could you explain the logic of the change here?
Also, if it's going to update health, then put it at the end of the damage procs.

**Comment by itseasytosee** [comment]
Very simple.
Slime people get hurt by toxin healing unless the force var in the adjust damage proc is set to true. 
Rust healing as a heretic is an effect that reconstructs your body though eldrich means, not a chemical reaction that slimes are adverse to, so it should heal them as normal.
So I set the force var to true so slime people heretics don't get hurt in a way that doesn't make sense by an effect that normally benefits them.

**Review by Rohesie — APPROVED** [review]
I was thinking it was the update health boolean, not the forced one.
Makes sense.

**Comment by Mothblocks** [comment]
@Rohesie That's why named arguments for booleans are important!

File: code/modules/antagonists/eldritch_cult/knowledge/rust_lore.dm [hunk 1]
```diff
@@ -66,7 +66,7 @@
 	var/mob/living/living_user = user
 	living_user.adjustBruteLoss(-2, FALSE)
 	living_user.adjustFireLoss(-2, FALSE)
-	living_user.adjustToxLoss(-2, FALSE)
+	living_user.adjustToxLoss(-2, FALSE, TRUE)
```
Comment (Mothblocks) [inline code/modules/antagonists/eldritch_cult/knowledge/rust_lore.dm:69]: Named argument.

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 2]
```diff
@@ -2559,7 +2559,7 @@
 		M.drowsyness = max(M.drowsyness - (5 * REM * delta_time), 0)
 		M.AdjustAllImmobility(-40 * REM * delta_time)
 		M.adjustStaminaLoss(-10 * REM * delta_time, FALSE)
-		M.adjustToxLoss(-2 * REM * delta_time, FALSE)
+		M.adjustToxLoss(-2 * REM * delta_time, FALSE, TRUE)
```
Comment (Mothblocks) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:2562]: Named argument.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by moocowswag** [comment]
I think if this is merged slime heretics will still die after they ascend, the extra brute/burn/toxyloss healing you get when you ascend seems to be its own seperate healing source that is activated on ascension.

And it doesnt looked like you have touched that healing.

**Pull request merged**